src/commands/source/download.rs

```diff
@@ -42,14 +42,30 @@ enum DownloadResult {
 }
 
 impl DownloadResult {
-    fn to_verb(&self) -> &str {
-        match &self {
-            DownloadResult::Error => "failed",
-            DownloadResult::Forced => "forced",
-            DownloadResult::Skipped => "skipped",
-            DownloadResult::Succeeded => "succeeded",
+    fn to_index(&self) -> usize {
+        self.clone() as usize
+    }
+}
+
+fn to_verb(index: &usize) -> Option<String> {
+    match index {
+        0 => Some("failed".to_string()),
+        1 => Some("forced".to_string()),
+        2 => Some("skipped".to_string()),
+        3 => Some("succeeded".to_string()),
+        _ => None,
+    }
+}
+
+fn count_occurences(results: Vec<Result<DownloadResult>>) -> [i32; 4] {
+    let mut r: [i32; 4] = [0, 0, 0, 0];
+    for i in results {
+        match i {
+            Ok(i) => r[i.to_index()] += 1,
+            Err(_) => r[0] += 1,
         }
     }
+    r
```

Comment:
Ugh, this unfortunately also seems super non-idiomatic.
Let's better have another look at it together to see if we can up with some nice and clean / idiomatic solution.